### PR TITLE
Extending ember-template-lint config type definition

### LINKED ISSUE #514: [enhancement] ember-template-lint additional types
I am trying to add a local plugin for ember-template lint that looks something like the following;

```js
const TEMPLATE_LINT_CONFIG: TemplateLintConfig = {
  plugins: [
    {
      name: 'local-rules',
      rules: {
        'hello-world': require(path.resolve(__dirname, '../../template-lint-rules/hello-world')),
      },
    },
  ],
  rules: {
    'hello-world': 'error'
  },
};
```

Looking at the types for ember-template-lint config it looks like it is missing some fields.

https://github.com/checkupjs/checkup/blob/9fe60a9f02ef9d9ef1d68b2f4be479e4d52aa15e/packages/core/src/types/ember-template-lint.ts#L29-L33

ember-template-lint doesn't expose types you can see the top level properties that are allowed here:

https://github.com/ember-template-lint/ember-template-lint/blob/b7195d0b6ae8051aff904a49bf9969741cfbf877/lib/get-config.js#L119-L126

> I am using @checkup/core@0.1.1

### PR DESCRIPTION
**Pull request by scalvert**
Fixes #514

### CODE REVIEW

**Comment by scalvert** [comment]
cc @gabrielcsapo

File: packages/core/src/types/ember-template-lint.ts [hunk 1]
```diff
@@ -27,7 +27,12 @@ export interface TemplateLintReport {
 }
 
 export interface TemplateLintConfig {
+  extends?: string[];
   rules?: {
     [name: string]: RuleLevel | RuleLevelAndOptions;
   };
+  pending?: string[];
+  ignore?: string[];
+  plugins?: string[];
```
Comment (rwjblue) [inline packages/core/src/types/ember-template-lint.ts:36]:
```ts
interface TemplateLintPluginObject {
  name: string;
  plugins: TemplateLintPlugins[];
  rules: {
    [ruleName: RuleName]: TemplateLintRuleDefinition;
  };
  configurations: {
    [configuration: string]: TemplateLintConfig;
  }
}
type TemplateLintPlugin = string | TemplateLintPluginObject;
```

```suggestion
  plugins?: TemplateLintPlugin[];
```

File: packages/core/src/types/ember-template-lint.ts [hunk 2]
```diff
@@ -27,7 +27,12 @@ export interface TemplateLintReport {
...
+  overrides?: string[];
```
Comment (rwjblue) [inline packages/core/src/types/ember-template-lint.ts:37]:
```ts
interface TemplateLintOverride {
  files: string[];
  rule: {
    [ruleName: RuleName]: RuleLevel | RuleLevelAndOptions; 
  }
}
```

```suggestion
  overrides?: TemplateLintOverride[];
```

File: packages/core/src/types/ember-template-lint.ts [hunk 3]
```diff
@@ -27,7 +27,12 @@ export interface TemplateLintReport {
 }
 
 export interface TemplateLintConfig {
+  extends?: string[];
   rules?: {
     [name: string]: RuleLevel | RuleLevelAndOptions;
```
[the remark this answers is not in the document]
Comment (scalvert) [reply packages/core/src/types/ember-template-lint.ts:32]: `An index signature parameter type cannot be a type alias`.

:(

**Pull request merged**